caveclient/materializationengine.py

```diff
@@ -275,11 +275,30 @@ def cg_client(self):
     def version(self) -> int:
         """The version of the materialization. Can be used to set up the
         client to default to a specific version when timestamps or versions are not
-        specified in queries. If not set, defaults to the most recent version."""
+        specified in queries. If not set, defaults to the most recent version.
+
+        Note that if this materialization client is attached to a CAVEclient,
+        the version must be set at the CAVEclient level.
+        """
+        if self.fc is not None and self.fc.version is not None:
+            return self.fc.version
         if self._version is None:
             self._version = self.most_recent_version()
         return self._version
 
+    @version.setter
+    def version(self, x: Optional[int]):
+        if self.fc is not None and self.fc.version is not None:
+            msg = (
+                "Cannot set `version` for materialization client when attached to a "
+                "CAVEclient with a version, set at the CAVEclient level instead."
+            )
+            raise ValueError(msg)
+        if int(x) in self.get_versions(expired=True):
```

Comment:
note this slight change here to allow expired versions, but I think we wanted this?

Reply:
Seems right to me — this code predates that option, I think, and never got updated.